### PR TITLE
Add support for ghostty terminal

### LINKED ISSUE #299: Support for images in Ghostty
This software worked fine if I forced it to `TERM=xterm-kitty` to get through the feature detection, but [Ghostty](https://mitchellh.com/ghostty) fully supports the kitty graphics protocol at minimum.

I'm not sure off hand if it could support jump marks, I'd have to experiment a bit.

But the following are appropriate for identifying Ghostty:

```
TERM_PROGRAM=ghostty
TERM=xterm-ghostty
```

I can share the whole terminfo if you'd like, but seems your detection is purely based on TERM and TERM_PROGRAM rather then querying the terminfo database.

### PR DESCRIPTION
**Pull request by mattrobenolt**
Fixes #299

### CODE REVIEW

**Comment by mattrobenolt** [comment]
Sorta orthogonal to this, longer term if you care, the kitty graphics protocol can be detected with a query, which can eliminate the need to explicitly support certain terminals, and this list becoming a maintainer burden. I'm not advocating for implementing this, just an FYI if you were unaware this existed. That way any new terminals, whether closed source, whatever, that implement this protocol can utilize the feature automatically.

https://sw.kovidgoyal.net/kitty/graphics-protocol/#querying-support-and-available-transmission-mediums

File: README.md [hunk 1]
```diff
@@ -20,7 +20,7 @@ mdcat in [WezTerm], with "One Light (base16)", "Gruvbox Light", and "Darcula
 
 ## Features
 
-`mdcat` works best with [iTerm2], [WezTerm], and [kitty], and a good terminal font with italic characters.
+`mdcat` works best with [iTerm2], [WezTerm], [kitty], and [Ghostty], and a good terminal font with italic characters.
```
Comment (swsnr) [inline README.md:23]: As a concession to my bad impression about the tool, let's not sort of advertise it here, at least not until it's really open source 🙂

Let's just add it to the table and call it a day 🙂

**Review by swsnr — CHANGES_REQUESTED** [review]
Looks good 🙏

Thanks for updating README and the manual page 🙏

Could you also add a CHANGELOG entry as well?

File: README.md (hunk 1 above)
Comment (mattrobenolt) [reply README.md]: Removed, left in the table.

**Comment by mattrobenolt** [comment]
> Could you also add a CHANGELOG entry as well?

Added.

**Comment by swsnr** [comment]
> Sorta orthogonal to this, longer term if you care, the kitty graphics protocol can be detected with a query, which can eliminate the need to explicitly support certain terminals, and this list becoming a maintainer burden.

I'm aware but it'd be quite a bit of an refactoring to turn detection from the purely passive process it is now to an active process, and it'd complicate things, as we'd have to deal with eg timeouts etc.

So far it hasn't been worth the effort 🤷

**Pull request merged**